### ISSUE TITLE
Timeshift for Radio channels

### ISSUE THREAD

**Issue by cuh7b5** [kodi-pvr/pvr.iptvsimple#510]
Hello,
On Arch linux, Kodi 19:
I have some mp3 radio streams in my m3u file like this:
```
#EXTINF:-1,tvg-logo="https://www.radio.net/images/broadcasts/88/e1/3244/1/c300.png" group-title="radio" radio="true",BBC Radio 2
http://stream.live.vc.bbcmedia.co.uk/bbc_radio_two

```
Thimeshift is enabled in the pvr client and works fine for TV streams. But for radio streams I get a strange behavior: I start the radio stream, leave it play for some seconds. Pressing the pause button then the play button, the stream starts playing from the beggining instead of pausing/resuming. Seeking forward and backward buttons result the same: the stream starts from the beggining (i.e. the point where I originally started the stream).

Is there a way to properly configure timeshift for radio channels?

Thanks!

**Comment by phunkyfish** [comment]
When I built the feature I tried it with a few radio channels and it worked. But I’ll be honest that the testing wasn’t extensive.

it might be that for certain audio streams it had issues.

**Comment by phunkyfish** [comment]
Do you have timeshift enabled for all streams?

**Comment by phunkyfish** [comment]
That is really odd. The controls that should be there on the OSD are missing. It's like there is no timeshift enabled at all. But clearly there is. Must be a bug in kodi.

**Comment by cuh7b5** [comment]
"Enable timeshift for all streams" is enabled in the PVR IPTV Simple client. I have all the OSD entries such as Channel backward, Forward, Slow down, Fast forward, Stop, Pause.

**Comment by phunkyfish** [comment]
This PR should correctly enable the buttons in Matrix: https://github.com/xbmc/xbmc/pull/19705 But it won't be released till 19.2.

Do you have an example stream I could test with in the meantime?

**Comment by cuh7b5** [comment]
I cannot attach m3u files here. But you can create one with this content:

```
#EXTINF:-1,tvg-logo="https://static.radio.net/images/broadcasts/27/22/12470/c300.png" group-title="radio" radio="true",LFM
https://lausannefm.ice.infomaniak.ch/lausannefm-high.mp3
#EXTINF:-1,tvg-logo="https://static.radio.net/images/broadcasts/89/d4/3214/1/c300.png" group-title="radio" radio="true",Radio France Internationale (RFI) Musique
http://live02.rfi.fr/rfimusiquemonde-96k.mp3
#EXTINF:-1,tvg-logo="https://www.radio.net/images/broadcasts/88/e1/3244/1/c300.png" group-title="radio" radio="true",BBC Radio 2
http://stream.live.vc.bbcmedia.co.uk/bbc_radio_two
#EXTINF:-1,tvg-logo="https://www.radio.net/images/broadcasts/4a/cd/11932/1/c300.png" group-title="radio" radio="true",BBC Radio 6 Music
http://stream.live.vc.bbcmedia.co.uk/bbc_6music
```

**Comment by phunkyfish** [comment]
Ok, I can reproduce the problem as you see it. I think the problem is that timeshift was designed to work with the VideoPlayer in kodi and I'm not sure it works the MusicPlayer/PAPlayer that is used for Radio.

**Comment by phunkyfish** [comment]
I have a workaround tbough. 

Remove `radio="true"` from the M3U and open it as a TV channel. Then timeshift will work. In the meantime I will see if it's possible to specify the player to be used for an inpustream as that should fix the problem.

**Comment by phunkyfish** [comment]
I also have a fix going to kodi to allow PVR add-ons to override the player used by an inputstream add-on. This could take a little while to go into kodi. Once in Kodi Nexus it would then need to backported to Matrix so would not be available to 19.4 comes out.

https://github.com/xbmc/xbmc/pull/20386

**Comment by phunkyfish** [comment]
This is fixed in the next Matrix nightly build. It will be included in 19.4 when that gets released.
